presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -687,6 +687,31 @@ public void testInsert()
         assertQueryFails("INSERT INTO test_insert (b) VALUES (ARRAY[1.23E1])", "line 1:37: Mismatch at column 1.*");
 
         assertUpdate("DROP TABLE test_insert");
+
+        assertUpdate("CREATE TABLE test_insert(a row(b int, c varchar))");
+        assertUpdate("INSERT INTO test_insert VALUES (row(1, '1001'))", 1);
+        assertUpdate("INSERT INTO test_insert VALUES row(2, '1002'), ((3, '1003'))", 2);
+        assertUpdate("INSERT INTO test_insert VALUES (4, '1004'), (row(5, '1005'))", 2);
```

Comment:
This test case passing seems a bit concerning because by the same logic

```
 assertUpdate("CREATE TABLE test_insert(a row(b int, c varchar), d int)"); 
 assertUpdate("INSERT INTO test_insert(a, d) VALUES (4, '1004', 6), (row(5, '1005'), 6)", 2);
```
should pass but it fails

Reply:
No, it shouldn't pass. In your test case, `(4, '1004', 6)` is explicitly different from `(row(4, '1004'), 6)` or `((4, '1004'), 6)`. Only when there is just one target column, could we omit the bracket for a row.

Reply:
Ohh so ((4, '1004'), 6) will work